test/fuzz/TrancheWrapper/PreviewMint.fuzz.t.sol

```diff
@@ -0,0 +1,63 @@
+/**
+ * Created by Pragma Labs
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity 0.8.22;
+
+import { TrancheWrapper_Fuzz_Test } from "./_TrancheWrapper.fuzz.t.sol";
+import { TrancheExtension } from "../../utils/extensions/TrancheExtension.sol";
+
+/**
+ * @notice Fuzz tests for the function "deposit" of contract "Tranche Wrapper".
+ */
+contract PreviewMint_TrancheWrapper_Fuzz_Test is TrancheWrapper_Fuzz_Test {
+    /* ///////////////////////////////////////////////////////////////
+                              SETUP
+    /////////////////////////////////////////////////////////////// */
+
+    function setUp() public override {
+        TrancheWrapper_Fuzz_Test.setUp();
+    }
+    /*//////////////////////////////////////////////////////////////
+                              TESTS
+    //////////////////////////////////////////////////////////////*/
+
+    function testFuzz_Success_previewMint_NonZeroSupply(
+        uint80 vas,
+        uint256 totalSupply,
+        uint128 totalAssets,
+        uint256 shares
+    ) public {
+        // Given: totalSupply is bigger than zero.
+        // And: no overflows.
+        totalSupply = bound(totalSupply, 1, type(uint256).max - vas);
+        if (uint256(totalAssets) + vas > 0) shares = bound(shares, 0, type(uint256).max / (uint256(totalAssets) + vas));
+
+        // And: Tranche state is set.
+        tranche = TrancheExtension(setTrancheState(vas, totalSupply, totalAssets));
+
+        // When: previewMint is called with shares.
+        uint256 actualAssets = tranche.previewMint(shares);
```

Comment:
calls should be done on trancheWrapper instead of tranche. As we are deploying a new tranche here, we should deploy a new trancheWrapper for that tranche and run tests on that trancheWrapper.